pages/treeherder.py

```diff
@@ -51,6 +59,42 @@ def open_perfherder_page(self):
         from perfherder import PerfherderPage
         return PerfherderPage(self.base_url, self.selenium).wait_for_page_to_load()
 
+    def select_next_job(self):
+        el = self.selenium.find_element(*self._resultset_locator)
+        Wait(self.selenium, self.timeout).until(EC.visibility_of(el))
+        el.send_keys(Keys.ARROW_RIGHT)
+        Wait(self.selenium, self.timeout).until(lambda s: self.job_result_status)
+        return self.selenium.find_element(*self._next_job_title).get_attribute('title')
+
+    def add_selected_job_to_pinboard(self):
+        el = self.selenium.find_element(*self._resultset_locator)
+        Wait(self.selenium, self.timeout).until(EC.visibility_of(el))
+        el.send_keys(Keys.SPACE)
+        Wait(self.selenium, self.timeout).until(lambda s: self.is_pinboard_open)
+
+    @property
+    def is_pinboard_open(self):
+        return self.is_element_visible(self._pinboard_locator)
+
+    @property
+    def pinned_job_title(self):
+        return self.selenium.find_element(*self._next_job_title).get_attribute('title')
+
+    def pin_job_from_logviewer(self):
+        self.select_next_job()
+        el = Wait(self.selenium, self.timeout).until(
+            EC.visibility_of_element_located(self._job_details_actionbar_locator))
+        el.find_element(*self._logviewer_pin_job_locator).click()
+        Wait(self.selenium, self.timeout).until(lambda s: self.is_pinboard_open)
+
+    def clear_pinboard(self):
+        self.selenium.find_element(*self._open_save_menu_locator).click()
+        self.selenium.find_element(*self._clear_all_menu_locator).click()
+
+    @property
+    def any_jobs_pinned(self):
+        return not self.is_element_visible(self._pinboard_item_locator)
```

Comment:
If we returned a total number of jobs in the pinboard, this could be used to assert that it's zero or non-zero.